### ISSUE TITLE
CocoaPods not installed. Skipping pod install. CocoaPods not installed or not in valid state.

### ISSUE THREAD

**Issue by Sbenari** [CocoaPods/CocoaPods#11290]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report
after working on my project for months without a hitch, suddenly out of the blue I get the `Warning: CocoaPods not installed. Skipping pod install `error. Following a pandemonium of instructions from various user groups such as StackOverflow and others. trying many different things I reached no resolution to this Major problem, which appears to be ignored by the developer of this system. why is this happening? and why there is no detailed step-by-step instruction on how to fix this problem? Hours and days are spent figuring this thing out. in my case without results. My project is built on macOS and I use Android Studio ArcticFox. 

## What did you do?

ℹ 1)` sudo gem uninstall cocoapods`
2)`sudo gem install cocoapods`
3)restart
4)Invalidate cache and restart
5)`pod Install` in the ios directory
6)restart
7)` brew install cocoapods`
8)restart
9) `flutter build`
10) restart
11) remove podfile.lock 
12)restart

none of these worked I have no idea what else I can do

## What did you expect to happen?

ℹ My app is to be installed onto the simulator.

## What happened instead?

ℹ ```
Warning: CocoaPods not installed. Skipping pod install.
  CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
  Without CocoaPods, plugins will not work on iOS or macOS.
  For more info, see https://flutter.dev/platform-plugins
To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

CocoaPods not installed or not in valid state.
Error launching application on iPhone 12 mini.
```

## CocoaPods Environment

ℹ ### Stack

```
   CocoaPods : 1.11.3
        Ruby : ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.x86_64-darwin21]
    RubyGems : 3.0.3.1
        Host : macOS 12.3 (21E230)
       Xcode : 13.3 (13E113)
         Git : git version 2.32.0 (Apple Git-132)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

**Comment by dnkoutso** [comment]
We cannot resolve issues that appear to be related to Flutter. Please file a new issue _without_ anything Flutter and a sample app that clearly demonstrates a CocoaPods gem issue.

The error message you see is coming from Flutter itself. There is no code in CocoaPods that displays the error message you pasted.

**Comment by Sbenari** [comment]
I've been there before, flutter people will put it on you, and you put them on Flutter. It's your issue as much as it's theirs!

**Comment by dnkoutso** [comment]
It is not though. The actual error message is coming from Flutter itself. Something in it is not detecting cocoapods being installed. If you run `pod install` in a brand new vanilla cocoapods project does it work? Perhaps it is an environment issue in which cocoapods is not installed or a new Ruby version where the gem is missing.

Either way it is extremely time consuming to diagnose those and so you must remove all potential 3rd party factors to show this is truly a CocoaPods issue before we can investigate it.

**Comment by Sbenari** [comment]
It is just as time-consuming for me to deal with your product, if you want to help, then by nature, it's a time-consuming ordeal. you should do all you can to have a reliable product running on whatever platform it's on. Helping is not only a quick fix thing! I have no resources and knowledge to deal with this major problem. why can't you at least leave this issue open so that someone else with knowledge of the matter can try to solve it?

**Comment by paulb777** [comment]
@Sbenari Part of solving the problem is isolating it to the root cause. In this case, the error message is from the Flutter product, not CocoaPods.

**Comment by dnkoutso** [comment]
I am kindly requesting you provide a sample app _without_ Flutter to further show or prove this is perhaps CocoaPods issue and not in Flutter. I insist this is either a Flutter issue or an environment and not a CocoaPods gem issue.

**Comment by Sbenari** [comment]
@dnkoutso this problem appears when I try to test my app on the Simulator, instead of installing the app, it errors with the cocoapods message. the problem is not with the app itself which runs fine when I use Xcode to install it into the simulator. the problem is when I try to install my app with Android studio.
